testcafe/tests/pages/drive-model-public.js

```diff
@@ -62,27 +86,50 @@ export default class PublicDrivePage {
       .expect(this.btnPublicMobileDownload.exists)
       .notOk('Create my Cozy Button (mobile) exists')
     await isExistingAndVisibile(this.logo, 'Logo')
-    await isExistingAndVisibile(this.toolbarFilesPublic, 'toolbarFiles')
+    if (type == 'file') {
+      await isExistingAndVisibile(this.toolbarFilesPublic, 'toolbarFiles')
+      await isExistingAndVisibile(
+        this.btnPublicCreateCozyFile,
+        'Create my Cozy Button'
+      )
+    } else if (type == 'folder') {
+      await isExistingAndVisibile(this.toolbarFolderPublic, 'toolbarFolder')
+      await isExistingAndVisibile(
+        this.btnPublicCreateCozyFolder,
+        'Create my Cozy Button'
+      )
+    }
     await isExistingAndVisibile(this.btnPublicDownload, 'Download FolderButton')
-    await isExistingAndVisibile(
-      this.btnPublicCreateCozy,
-      'Create my Cozy Button'
-    )
   }
 
-  async checkActionMenuPublicMobile() {
-    await t //Desktop element don't exists
-      .expect(this.logo.exists)
-      .notOk('Logo exists')
-      .expect(this.toolbarFilesPublic.exists)
-      .notOk('toolbar_file exists')
+  // @param {string} type : 'file' or 'folder' : the toolbar is different depending on share type
+  async checkActionMenuPublicMobile(type) {
+    await t
       .expect(this.btnPublicDownload.exists)
       .notOk('Download Button (desktop) exists')
-      .expect(this.btnPublicCreateCozy.exists)
-      .notOk('Create Cozy button (desktop) exists')
 
-    await isExistingAndVisibile(this.btnPublicMoreMenu, '[...] Button')
-    await t.click(this.btnPublicMoreMenu)
+    if (type == 'file') {
+      await t
+        .expect(this.logo.visible)
+        .notOk('Logo is visible') //logo still exist on mobile, but is not visible
+        .expect(this.toolbarFilesPublic.exists)
+        .notOk('toolbar_file exists')
+        .expect(this.btnPublicCreateCozyFile.exists)
+        .notOk('Create Cozy button (desktop) exists')
+      await isExistingAndVisibile(this.btnPublicMoreMenuFile, '[...] Button')
+      await t.click(this.btnPublicMoreMenuFile)
+    } else if (type == 'folder') {
+      await t
+        .expect(this.logo.exists)
+        .notOk('Logo exists')
+        .expect(this.toolbarFolderPublic.exists)
+        .notOk('toolbar_file exists')
+        .expect(this.btnPublicCreateCozyFolder.exists)
+        .notOk('Create Cozy button (desktop) exists')
+      await isExistingAndVisibile(this.btnPublicMoreMenuFolder, '[...] Button')
+      await t.click(this.btnPublicMoreMenuFolder)
```

Comment:
Are you sure you need to split it that way? It seems that except the first expect the rest is the same?

Reply:
see https://github.com/cozy/cozy-drive/pull/1433/files#r257228422